src/vs/workbench/browser/actions/layoutActions.ts

```diff
@@ -514,6 +514,35 @@ export class ToggleTabsVisibilityAction extends Action2 {
 }
 registerAction2(ToggleTabsVisibilityAction);
 
+// --- Toggle Pinned Tabs On Separate Row
+
+export class ToggleSeperatePinnedTabsAction extends Action2 {
+
+	static readonly ID = 'workbench.action.toggleSeperatePinnedTabs';
+
+	constructor() {
+		super({
+			id: ToggleSeperatePinnedTabsAction.ID,
+			title: {
+				value: localize('toggleSeperatePinnedTabs', "Seperate Pinned Tabs"),
+				original: 'Seperate Pinned Tabs'
+			},
+			category: Categories.View,
```

Comment:
We should not offer this action when tabs are disabled, since it only works then. There is a `precondition` property that can have a context key expression.